include/net/tftp.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file tftp.h
+ *
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_TFTP_H_
+#define ZEPHYR_INCLUDE_NET_TFTP_H_
+
+#include <zephyr.h>
+#include <net/socket.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct tftpc {
+	u8_t   *user_buf;
+	u32_t  user_buf_size;
+};
+
+/* Name: tftp_get
```

Comment:
as a public API, you'll need to properly document it via doxygen format (@brief, @param etc...)

Reply:
Yes, I was wondering how the documentation part will be done.. Thanks for pointing this out.. 

BTW - Do you have any thoughts on the API, specifically the IO params ? Obviously the remote file has to a string but we could've kept the local file a string or an array. Though the former makes more sense (in the TFTP context) but it requires access to the File System+NVS and we wanted to keep our solution simple. Therefore we allowed the user to simply provide an array which can be populated by the stack.

Reply:
> BTW - Do you have any thoughts on the API, specifically the IO params ?

FYI, there's https://github.com/zephyrproject-rtos/zephyr/issues/13212 RFC for requirements on implementing higher-level networking protocols. In summary, it argues a) for always consider and support a "streaming" usecase (e.g., here transferring 1GB over TFTP and wrting it to an external flash); b) structure APIs calls as consistent as possible to standard Socket API, i.e. center around read(void *your_protocol, void *buf, size_t sz) and write(void *your_protocol, const void *buf, size_t sz) calls.

> but it requires access to the File System+NVS and we wanted to keep our solution simple.

Surely, there's no need for a networking protocol implementation to depend on a particular FS implementation. Users who need to work with an FS, can implement integration on their app side.